### PR TITLE
add support for the 10 digits for text manipulation

### PR DESCRIPTION
**Pull request by alexboche**
Also update documentation accordingly.

### CODE REVIEW

File: castervoice/lib/ccr/core/text_manipulation.py [hunk 1]
```diff
@@ -14,6 +14,8 @@
 from castervoice.lib.ccr.core.punctuation import text_punc_dict,  double_text_punc_dict
 from castervoice.lib.alphanumeric import caster_alphabet
 
+base_number_dict = {"zero": "0", "one": "1", "two": "2", "three": "3", "four": "4", "five": "5", "six": "6", "seven": "7", "eight": "8", "nine": "9" }
+number_dict = {"num "+k:base_number_dict[k] for k in base_number_dict}
```
Comment (kendonB) [inline castervoice/lib/ccr/core/text_manipulation.py]: should be numb here.

Comment (kendonB) [reply castervoice/lib/ccr/core/text_manipulation.py]: it seems to get misinterpreted as dictation less often when using numb for whatever reason

**Comment by kendonB** [comment]
@comodoro is the testing system ready to test something like this? In pseudo code:

```
original_string = "The digit 2 will be replaced by 3"
command_to_test = "replace ross numb two with numb three" # assumes cursor is at the start of the string
new_string = run_command_as_a_text_manipulation(input_string = original_string, command = command_to_test)
assert_string(new_string, "The digit 3 will be replaced by 3")
```

**Comment by comodoro** [comment]
Not at the moment, although it is possible to add functional tests like this separately from existing tests.

**Comment by kendonB** [comment]
@comodoro this is the kind of thing where I really have no idea where to start to do `run_command_as_a_text_manipulation`. Any tips on getting started?

**Comment by comodoro** [comment]
I am on a shaky ground myself. Most projects I have encountered prefer unit or unit-like tests and I can see why, but testing functionalities seems quite important.

I would probably create parallel testing structure to existing (and refactor) and use Notepad and mimic for testing text functionality. I might try to cobble something together Sunday.

**Comment by LexiconCode** [comment]
> 
> 
> I am on a shaky ground myself. Most projects I have encountered prefer unit or unit-like tests and I can see why, but testing functionalities seems quite important.
> 
> I would probably create parallel testing structure to existing (and refactor) and use Notepad and mimic for testing text functionality. I might try to cobble something together Sunday.

Yes unit tests are going to become even more important as we become cross-platform.

**Comment by kendonB** [comment]
Are we able to use a different editor to notepad for these simulations? Notepad has inconsistent cursor behavior: https://github.com/texstudio-org/texstudio/issues/599#issuecomment-493288364

Notepad++ is fast to open and reliable. A perfect system would test across multiple editors though...

**Comment by kendonB** [comment]
FYI, I'm going to wait until after tomorrow to merge this to see if @comodoro comes through with some clever testing infrastructure for this sort of thing. If he does, then we'll (Alex and I) will take the opportunity to learn how to put together some tests for the case of changing text around by voice.

**Comment by LexiconCode** [comment]
let me know when it's ready to merge..

**Comment by kendonB** [comment]
I'm going to merge this one and open an issue for setting up tests for text manipulations by simulating a text editor.

**Pull request merged**